tests/functional/flakes/flakes.sh

```diff
@@ -220,6 +220,13 @@ nix store gc
 nix registry list --flake-registry "file://$registry" --refresh | grepQuiet flake3
 mv "$registry.tmp" "$registry"
 
+# Ensure that locking ignores the user registry.
+mkdir -p "$TEST_HOME/.config/nix"
+ln -sfn "$registry" "$TEST_HOME/.config/nix/registry.json"
+nix flake metadata --flake-registry '' flake1
```

Comment:
@edolstra it looks like the test was not able to capture the previous regression. Do you know why this is the case?

Reply:
Ah, We actually tried to use a flake from the registry. So this usecase is not tested in this pull request.